starter_code/src/movies.js

```diff
@@ -1,15 +1,166 @@
 /* eslint no-restricted-globals: 'off' */
 
+
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+//..sort by year
+function orderByYear(array) {
+    const newMoviesArray = JSON.parse(JSON.stringify(array))
+    return orderByTitle(array).sort((a, b) => a["year"] - b["year"]);
+}
+
+//..sort by title (only if same year)
+function orderByTitle(array) {
+    const newMoviesArray = JSON.parse(JSON.stringify(array))
+    return newMoviesArray.sort(function(a, b) {
+      if (a["title"] < b["title"]) {
+        return -1;
+      } 
+      if (a["title"] > b["title"]) {
+        return 1;
+      }
+      if (a["title"] == b["title"]) {
+        return 0;
+      }
+    }); 
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+function theSpielbergMovies(array) {
+    const newMoviesArray = JSON.parse(JSON.stringify(array));
+    const spielbergMovies = newMoviesArray.filter(theDirector => theDirector["director"] == "Steven Spielberg");
+    return spielbergMovies;
+}
+  
+function theDramaMovies(array) {
+    const newMoviesArray = JSON.parse(JSON.stringify(array));
+    const theDramaMovies = newMoviesArray.filter(theDrama => theDrama["genre"].includes("Drama"));
+    return theDramaMovies;
+}
+
+function howManyMovies(array) {
+    const dramaArray = theDramaMovies(array);
+    return theSpielbergMovies(dramaArray).length;
+}
 
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+//..cut down to 20
+function cutArrayToTwenty(array) {
+  const newMoviesArray = JSON.parse(JSON.stringify(array))
+  if (newMoviesArray.length > 20) {
+    return newMoviesArray.slice(0,20);
+  } else {
+    return newMoviesArray;
+  }
+}
+
+//..Get titles only
+function getTitleOnly(array) {
+  const newMoviesArray = JSON.parse(JSON.stringify(array));
+  const titlesArray = newMoviesArray.map(theTitle => {
+    return theTitle["title"]
+  })
+  return titlesArray
+}
+
+function orderAlphabetically(array) {
+  let twentyMovies = orderByTitle(array);
+  twentyMovies = cutArrayToTwenty(twentyMovies);
+  twentyMovies = getTitleOnly(twentyMovies);
+  return twentyMovies;
+}
 
```

Comment:
Nicely done. Another way you can write your function is:
function orderAlphabetically(array) {
  return getTitleOnly(cutArrayToTwenty(orderByTitle(array)))
}